File: dotnet/Lexy.Poc/Lexy.Poc.Core/Specifications/IScenarioRunner.cs
using Lexy.Poc.Core.Language;
using Lexy.Poc.Core.Parser;
using Microsoft.Extensions.DependencyInjection;

namespace Lexy.Poc.Core.Specifications
{
    public interface IScenarioRunner
    {
        bool Failed { get; }
        Scenario Scenario { get; }

        void Initialize(string fileName, Nodes nodes, Scenario scenario,
            ISpecificationRunnerContext context, IServiceScope serviceScope, IParserLogger parserLogger);

        void Run();
        string ParserLogging();
    }
}

File: typescipt/src/specifications/iScenarioRunner.ts


export interface IScenarioRunner {
   boolean Failed
   Scenario Scenario

   void Initialize(string fileName, RootNodeList rootNodeList, Scenario scenario,
     ISpecificationRunnerContext context, IServiceScope serviceScope, IParserLogger parserLogger);

   void Run();
   string ParserLogging();
}
